pkgs/cupertino_http/example/lib/main.dart

```diff
@@ -49,88 +45,71 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  List<Book> _books = [];
+  List<Book>? _books;
 
   @override
   void initState() {
     super.initState();
   }
 
-  void _runSearch(String query) {
+  // Get the list of books matching `query`.
+  // The `get` call will automatically use the `client` configurated in `main`.
+  Future<List<Book>> _getBooks(String query) async {
+    final response = await get(
+      Uri.https(
+        'www.googleapis.com',
+        '/books/v1/volumes',
+        {'q': query, 'maxResults': '40', 'printType': 'books'},
+      ),
+    );
+
+    final json = jsonDecode(utf8.decode(response.bodyBytes)) as Map;
+    return Book.listFromJson(json);
```

Comment:
Nice! I was annoying how much tedious parsing was getting in the way of my example ;-)